src/SourceLocator/Type/MemoizingSourceLocator.php

```diff
@@ -58,7 +59,7 @@ public function locateIdentifiersByType(Reflector $reflector, IdentifierType $id
     private function reflectorCacheKey(Reflector $reflector) : string
     {
         return 'type:' . get_class($reflector)
-            . '#oid:' . spl_object_hash($reflector);
+            . '#oid:' . (\PHP_MAJOR_VERSION > 7 || \PHP_MINOR_VERSION > 1 ? spl_object_id($reflector) : spl_object_hash($reflector));
```

Comment:
Can you just use function_exists?

Reply:
no - const can be resolved on compile time and the code should be dropped once PHP 7.1 support is dropped

Reply:
I don't care about "can be resolved on compile time". I care about that code being understood by PHPStan.

Reply:
what is wrong with constants?